applications/machine_learning/src/modules/ml_runner.c

```diff
@@ -102,21 +213,16 @@ static bool event_handler(const struct event_header *eh)
 		return handle_sensor_event(cast_sensor_event(eh));
 	}
 
-	if (is_module_state_event(eh)) {
-		const struct module_state_event *event = cast_module_state_event(eh);
-
-		if (check_state(event, MODULE_ID(main), MODULE_STATE_READY)) {
-			__ASSERT_NO_MSG(state == STATE_DISABLED);
+	if (is_ml_result_event(eh)) {
+		return handle_ml_result_event(cast_ml_result_event(eh));
+	}
 
-			if (!init()) {
-				module_set_state(MODULE_STATE_READY);
-				state = STATE_ACTIVE;
-			} else {
-				report_error();
-			}
-		}
+	if (ML_STATE_CONTROL && is_ml_state_event(eh)) {
```

Comment:
can we write it as
```
if (CONDITION &&
    is_ml_state_event(eh)) {
...
```

I find it easier to follow as you don't have to look right to see what event is handled in this block.

Reply:
Changed.